### PR TITLE
Remove self-cast for each instance

### PR DESCRIPTION
**Pull request by junjihashimoto**

### CODE REVIEW

File: ffi/src/ATen/Cast.hs [hunk 1]
```diff
@@ -36,155 +36,7 @@ import Foreign.Ptr
...
-instance Castable Word16 Word16 where
-  cast x f = f x
-  uncast x f = f x
-
-instance Castable Word32 Word32 where
-  cast x f = f x
-  uncast x f = f x
-
-instance Castable Word64 Word64 where
+instance Castable a a where
```
Comment (apaszke) [inline ffi/src/ATen/Cast.hs:39]: Don't you have to add `{-# OVERLAPPABLE #-}` here?

**Review by apaszke — APPROVED** [review]

File: ffi/src/ATen/Cast.hs (hunk 1 above)
Comment (apaszke) [reply ffi/src/ATen/Cast.hs:39]: Seems like it works out nicely! There's only one extra self-cast I had to add some time ago that you missed and that's causing the compile error.

Comment (junjihashimoto) [reply ffi/src/ATen/Cast.hs:39]: When we want to overlap self-cast of a instance, we have to add `{-# OVERLAPPABLE #-}`.
For now, I feel the overlap-pragma is not needed.

**Review by apaszke — APPROVED** [review]

**Comment by junjihashimoto** [comment]
thank you for reviewing.

**Pull request merged**